fc/mef.py

```diff
@@ -217,265 +52,168 @@ def clustering_gmm(data, n_clusters = 8, initialization = 'distance_sub',
-    # mu = np.mean(data)
-    sigma = np.std(data)
-    # Calculate histogram
-    hist, __ = np.histogram(data, bin_edges)
-    # Do Gaussian blur on histogram
-    # We have found empirically that using one half of the distribution's 
-    # standard deviation results in a nice fit.
-    hist_smooth = scipy.ndimage.filters.gaussian_filter1d(hist, sigma/2.)
-    # Extract peak
-    i_max = np.argmax(hist_smooth)
-    peak = bin_centers[i_max]
-
-    return peak, hist_smooth
-
-def find_peaks_median(data):
-    """
-    Find the median of a data array.
-
-    Parameters
-    ----------
-    data : array
-        Nx1 array to calculate the median from.
-
-    Returns
-    -------
-    peak : int or float
-        Median of `data`.
-
-    """
-    peak = np.median(data)
-    return peak
-
-def select_peaks_proximity(peaks_ch,
-                           peaks_mef,
-                           peaks_ch_std,
-                           peaks_ch_std_mult_l = 2.5,
-                           peaks_ch_std_mult_r = 2.5,
-                           peaks_ch_min = 0,
-                           peaks_ch_max = 1023):
-    """
-    Select bead subpopulations based on proximity to a minimum and maximum.
-
-    This function discards some values from `peaks_ch` if they're closer
-    than `peaks_ch_std_mult_l` standard deviations to `peaks_ch_min`, or
-    `peaks_ch_std_mult_r` standard deviations to `peaks_ch_max`. Standard
-    deviations should be provided in `peaks_ch_std`. Then, it discards the
-    corresponding values in `peaks_mef`. Finally, it discards the values in
-    `peaks_mef` that have an undetermined value (NaN), and the
-    corresponding values in peaks_ch.
-
-    Parameters
-    ----------
-    peaks_ch : array
-        Sorted fluorescence values of bead populations in channel units.
-    peaks_mef : array
-        Sorted fluorescence values of bead populations in MEF units.
-    peaks_ch_std_mult_l, peaks_ch_std_mult_r : float, optional
-        Number of standard deviations from `peaks_ch_min` and
-        `peaks_ch_max`, respectively, that a value in `peaks_ch` has to be
-        closer than to be discarded.
-    peaks_ch_min, peaks_ch_max : int, optional
-        Minimum and maximum tolerable fluorescence value in channel units.
-
-    Returns
-    -------
-    sel_peaks_ch : array
-        Selected fluorescence values of bead populations in channel units.
-    sel_peaks_mef : array
-        Selected fluorescence values of bead populations in MEF units.
-
-    """
-    # Minimum peak standard deviation will be 1.0
+    # Default thresholds
+    if low is None:
+        if hasattr(populations[0], 'domain'):
+            low = 0.015*populations[0].domain(0)[0]
+        else:
+            raise TypeError("argument 'low' not specified")
+    if high is None:
+        if hasattr(populations[0], 'domain'):
+            high = 0.985*populations[0].domain(0)[-1]
+        else:
+            raise TypeError("argument 'high' not specified")
```

Comment:
I think referencing `populations[0]` will throw an error if `populations` is an empty list, which is never checked for. Not sure if we care about that edge case, though.